Sources/PackageGraph/RepositoryPackageContainerProvider.swift

```diff
@@ -49,6 +47,14 @@ public class RepositoryPackageContainerProvider: PackageContainerProvider {
         // Create the container wrapper.
         return RepositoryPackageContainer(identifier: identifier, repository: repository, manifestLoader: manifestLoader)
     }
+
+    public func mayNeed(containers: [Container.Identifier]) {
+        // Make an async lookup request to repository manager. This will kick off a clone of
+        // repo, if it is not present.
+        for identifier in containers {
+            repositoryManager.lookup(repository: identifier) { _ in }
+        }
```

Comment:
Let's land infrastructure for enabling this separate for commit to start using it, just to decouple the risk.